channels/plugins.py

```diff
@@ -12,6 +12,29 @@
     ChannelTopicDetail,
     ChannelUnitDetail,
 )
+from learning_resources.models import LearningResource
+
+
+def unpublish_topics_for_resource(resource):
+    """
+    Unpublish channels for topics that are used exclusively by the resource
+
+    Args:
+        resource(LearningResource): The resource that was unpublished
+    """
+    other_published = LearningResource.objects.filter(
+        published=True,
+    ).exclude(id=resource.id)
+
+    channels = Channel.objects.filter(
+        topic_detail__topic__in=resource.topics.all(),
+        channel_type=ChannelType.topic.name,  # Redundant, but left for clarity
```

Comment:
it seems like a topic can potentially have multiple channel topic details associated. is there a case where we would want more than one ChannelTopicDetail associated with a topic?

Reply:
The models currently support that, but I don't really think they should: departments, topics, units should have exactly 1 channel, which—as of this PR—might be published or unpublished.

Some releated slack discussion:
- A question by Nathan https://mitodl.slack.com/archives/C03K5HYGPT9/p1721667634521809
- A followup comment with thread attached https://mitodl.slack.com/archives/C03K5HYGPT9/p1721668541486719